tests/integration/long/test_ssl.py

```diff
@@ -108,4 +132,109 @@ def test_can_connect_with_ssl_ca(self):
         statement.consistency_level = ConsistencyLevel.ANY
         session.execute(statement)
 
-        cluster.shutdown()
\ No newline at end of file
+        cluster.shutdown()
+
+
+class SSLConnectionAuthTests(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        setup_cluster_ssl(client_auth=True)
+
+    def test_can_connect_with_ssl_client_auth(self):
+        """
+        Test to validate that we can connect to a C* cluster that has client_auth enabled.
+
+        This test will setup and use a c* cluster that has client authentication enabled. It will then attempt
+        to connect using valid client keys, and certs (that are in the server's truststore), and attempt to preform some
+        basic operations
+        @since 2.7.0
+
+        @expected_result The client can connect via SSL and preform some basic operations
+
+        @test_category connection:ssl
+        """
+
+        # Need to get absolute paths for certs/key
+        abs_path_ca_cert_path = os.path.abspath(CLIENT_CA_CERTS)
+        abs_driver_keyfile = os.path.abspath(DRIVER_KEYFILE)
+        abs_driver_certfile = os.path.abspath(DRIVER_CERTFILE)
+
+        tries = 0
+        while True:
```

Comment:
Curious why we need timeouts here. Could we improve by making sure the server is up, and expecting this to work first time?